check-ntservice/lib/check_ntservice_test.go

```diff
@@ -54,3 +77,41 @@ func TestNtService(t *testing.T) {
 		}
 	}
 }
+
+func TestRun(t *testing.T) {
+	testCases := []struct {
+		casename      string
+		cmdline       []string
+		expectStatus  checkers.Status
+		expectMessage string
+	}{
+		{
+			casename:      "check about running service",
+			cmdline:       []string{"-s", "running-service"},
+			expectStatus:  checkers.OK,
+			expectMessage: "",
+		},
+		{
+			casename:      "check about stopped service",
+			cmdline:       []string{"-s", "stopped-service"},
+			expectStatus:  checkers.CRITICAL,
+			expectMessage: "stopped-service-name: stopped-service-caption - Stopped",
+		},
+		{
+			casename:      "check about running service with exclude option",
+			cmdline:       []string{"-s", "service", "-E", "stopped"},
+			expectStatus:  checkers.OK,
+			expectMessage: "",
+		},
+	}
+
+	mockServiceState()
```

Comment:
This test had better reset `getServiceStateFunc` to initial state. Because tests that run after this test could be able to break.

For example:

```go
originalFunc := getServiceStateFunc
defer func(){
    getServiceStateFunc = originalFunc
}()
mockServiceState()
```

Reply:
That's true! I fix it, effeb49 .